pkg/query-service/app/integrations/messagingQueues/kafka/sql.go

```diff
@@ -74,3 +74,67 @@ ORDER BY
 `, start, end, queueType, topic, partition, timeRange)
 	return query
 }
+
+func generateNetworkLatencyThroughputSQL(start, end int64, consumerGroup, queueType string) string {
+	query := fmt.Sprintf(`
+--- Subquery for RPS calculation, desc sorted by rps
+SELECT
+    stringTagMap['messaging.client_id'] AS consumer_id,
+	stringTagMap['service.instance.id'] AS instance_id,
+    serviceName,
+    count(*) / ((%d - %d) / 1000000000) AS rps  -- Convert nanoseconds to seconds
+FROM signoz_traces.signoz_index_v2
```

Comment:
We need to make the same changes are as cosumer/producer 1. distributed table 2. if possible do the time denominator in code itself.